BeachHacks/Spotify-Queue-Web-2023
# [UPDATE/ADD] Updated backend Spotify authorization to handle user pla…

…yback scopes. Added front-end pages for administrative authorization control.

This PR reverts the client credential authorization model back to the previous authorization code flow. 
1a) Confirm Client-ID, Client-Secret, and AUTH-URL (the current one uses James' Spotify developer account).
1b) Depending on the used Client-ID & Client-Secret, you will need to update the Redirect URIs in the Spotify Developer Dashboard to whitelist the following URI -> http://localhost:3000/auth


Testing the application now is as follows:
1) Start client on http://localhost:3000 , and start server on http://localhost:3001.
2) Go to http://localhost:3000/admin . Click button Sign into Spotify. This authorizes the backend to use the signed in user for all Spotify API related functions, i.e. search and playback. 
3) You are redirected to confirmed Authorization page. Click button to go back to Dashboard. Functionality should resume as normal.

In this implementation, giving the backend access to a premium Spotify account requires visiting a special link from the front-end [http://localhost:3000/admin]. This can be changed.
